numpy/lib/tests/test_shape_base.py

```diff
@@ -27,6 +27,37 @@ def test_3d(self):
         assert_array_equal(apply_along_axis(np.sum, 0, a),
                            [[27, 30, 33], [36, 39, 42], [45, 48, 51]])
 
+    def test_preserve_subclass(self):
+        def double(row):
+            return row * 2
+        m = np.matrix(np.arange(4).reshape((2, 2)))
```

Comment:
Just a minor style thing -- I think this test would be clearer as just `np.matrix([[0, 1], [2, 3]])`, without the reshape.